org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/semantictokens/SemanticTokensVisitor.java

```diff
@@ -127,6 +129,52 @@ private void addToken(ASTNode node, TokenType tokenType, TokenModifier[] modifie
 		tokens.add(token);
 	}
 
+	/**
+	* Adds a semantic token to the list of tokens being collected by this
+	* semantic token provider. Overload for {@link #addToken(ASTNode, TokenType, TokenModifier[])}
+	* that adds no modifiers to the semantic token.
+	*
+	* @param node The AST node representing the location of the semantic token.
+	* @param tokenType The type of the semantic token.
+	*
+	* @apiNote This method is order-dependent because of {@link #encodeTokens()}.
+	* If semantic tokens are not added in the order they appear in the document,
+	* the encoding algorithm might discard them.
+	*/
+	private void addToken(ASTNode node, TokenType tokenType) {
+		addToken(node, tokenType, new TokenModifier[0]);
+	}
+
+	@Override
+	public boolean visit(QualifiedName node) {
+		if (node.resolveBinding() instanceof IPackageBinding) {
```

Comment:
Can add `|| node.resolveBinding() instanceof ITypeBinding` to cover more cases. E.g.

```java
import org.jcp.*;
import org.ietf.Eskibear;
```